src/transformers/trainer.py

```diff
@@ -1831,6 +1831,7 @@ def _inner_training_loop(
                     # AT THE VERY END!
                     _ = list(train_dataloader.sampler)
 
+        total_steps = 0
```

Comment:
let me know if you prefer to start with -1, so it'll be in sync with `step` which starts counting from -1.

Reply:
The step=-1 is mostly to make sure that variable is initialized even if we don't get into the loop for an epoch, so don't worry about it. This is fine.